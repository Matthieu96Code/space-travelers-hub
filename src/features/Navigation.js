import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/planet.png';

const Navigation = () => {
  const title = "Space Travelers' Hub";

  return (
    <nav className="navbar navbar-expand-lg bg-body-tertiary">
      <div className="container py-2">
        <a className="navbar-brand d-flex align-items-center gap-3 fw-semibold" href="/">
          <img src={logo} width="50px" alt={title} />
          <span className="d-none d-sm-none d-md-block d-lg-block">{title}</span>
        </a>
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse justify-content-end" id="navbarSupportedContent">
          <ul className="navbar-nav mb-2 mb-lg-0 gap-2 gap-sm-2 gap-md-2 gap-lg-4">
            <li className="nav-link">
              <NavLink to="rockets"> Rockets</NavLink>
            </li>

            <li className="nav-link">
              <NavLink to="missions"> Missions</NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Navigation;
