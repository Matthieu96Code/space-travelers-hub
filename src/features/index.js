export { default as Navigation } from './Navigation';
export { default as MissionsTable } from './missions/MissionsTable';
export { default as RocketsList } from './rockets/RocketsList';
