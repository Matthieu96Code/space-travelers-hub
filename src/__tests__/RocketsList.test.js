/* eslint-disable no-unused-vars */
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { RocketsList } from '../features';

const mockStore = configureMockStore();
const store = mockStore({
  rocket: {
    name: 'rocket',
    rockets: [
      {
        Active: false,
        boosters: 0,
        company: 'SpaceX',
        cost_per_launch: 6700000,
        country: 'Republic of the Marshall Islands',
        description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.',
        first_flight: '2006-03-24',
        flickr_images: ['https://imgur.com/DaCfMsj.jpg', 'https://imgur.com/azYafd8.jpg'],
        id: 1,
        reserved: false,
        rocket_id: 'falcon1',
        rocket_name: 'Falcon 1',
        rocket_type: 'rocket',
        stages: 2,
        success_rate_pct: 40,
        wikipedia: 'https://en.wikipedia.org/wiki/Falcon_1',
      },
    ],
  },
});

describe('<Rockets List>', () => {
  it('component should render correctly', () => {
    render(
      <Provider store={store}>
        <RocketsList />
      </Provider>,
    );

    const RocketsDivElement = screen.getByText('Falcon 1');
    expect(RocketsDivElement).toBeInTheDocument();
  });

  it('render list of rockets', async () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RocketsList />
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
