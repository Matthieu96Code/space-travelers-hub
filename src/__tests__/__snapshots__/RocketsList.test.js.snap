// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rockets List> render list of rockets 1`] = `
<div
  className="rockets-list"
>
  <div
    className="row py-3"
  >
    <div
      className="col-4"
    >
      <img
        alt="Falcon 1"
        className="img-fluid"
        src="https://imgur.com/DaCfMsj.jpg"
      />
    </div>
    <div
      className="col"
    >
      <h1
        className="fs-2"
      >
        Falcon 1
      </h1>
      <div>
        <p>
          <span />
          The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
        </p>
      </div>
      <button
        className="btn btn-primary btn-sm"
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
